.github/workflows/conda.yml

```diff
@@ -26,22 +26,28 @@ jobs:
     name: Build (and upload)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3.3.0
+      - name: Checkout source
+        uses: actions/checkout@v3
```

Comment:
Only major version, aligning to upstream best practice

Reply:
Where are these best practices documented? I actually prefer pinning explicitly.

We have dependabot to ensure we're running on most recent versions but with dependabot it is an explicit action

e.g. https://github.com/dask/distributed/pull/7464

Reply:
All other dependencies in our CI workflow (that is, the python packages) are automatically upgraded to the latest available version and we intervene to fix the problem if something goes wrong.

The best practice is taken from the README.md of the relevant actions:
- https://github.com/actions/checkout
- https://github.com/conda-incubator/setup-miniconda

from which I personally infer that API backwards compatibility won't be broken overnight as long as you pin the major version.